### PR TITLE
#274 Add doc strings

### PR DESCRIPTION
**Pull request by eelvira**
Adding docstrings and refactoring.

##Reference to official issue
#274 

## CLA
- [x] I have signed the CLA; if other committers are in the commit history, they have signed the CLA as well

### CODE REVIEW

File: prediction/src/preprocess/improved_lung_segmentation.py [hunk 1]
```diff
@@ -223,33 +242,40 @@ def grow_lungs(patient, seeds):
...
-            max_label_size = len(labeled_matrix[labeled_matrix == i])
-    for i in range(1, label_num + 1):
-        if label_size(labeled_matrix, i) < .2 * max_label_size:
-            labeled_matrix[labeled_matrix == i] = 0
-        else:
-            new_label_num += 1
-            labeled_matrix[labeled_matrix == i] = new_label_num
-    return new_label_num
+    bins = np.bincount(labeled_matrix.flatten())[1:]
+    return (bins >= .2 * bins.max()).sum()
```
Comment (reubano) [inline prediction/src/preprocess/improved_lung_segmentation.py]: Would you mind submitting this change in a separate PR?

File: prediction/src/preprocess/improved_lung_segmentation.py [hunk 2]
```diff
@@ -142,26 +160,26 @@ def extract_seeds(patient):
 def growing_bronchis(patient, seeds, initial_threshold=INITIAL_THRESHOLD,
                      step=STEP,
                      full_extraction=True):
-    """Extraction of large airways.
+    """
+    Extraction of large airways.
 
     Before the lungs are segmented, the trachea and
-    the bronchi are extracted using the technique of a region growing.
+    the bronchi are extracted using the tec hnique of a region growing.
```
Comment (reubano) [inline prediction/src/preprocess/improved_lung_segmentation.py]: Errant space? :)

File: prediction/src/preprocess/improved_lung_segmentation.py [hunk 3]
```diff
@@ -223,33 +242,40 @@ def grow_lungs(patient, seeds):
...
+    Args:
+        mask (np.ndarray[int|bool])
+
+    Returns:
+        int, the amount of colours in the re-labeled array
+    """
     mask, count = skimage.measure.label(mask, connectivity=1, return_num=True)
-    count = remove_trash(mask, count)
-    return count != 1
+    return remove_trash(mask) != 1
```
Comment (reubano) [inline prediction/src/preprocess/improved_lung_segmentation.py]: Would you mind submitting this change in a separate PR?

**Comment by reubano** [comment]
Nice work!

**Comment by eelvira** [comment]
@reubano, thank you for reviewing, done and rebased!

**Comment by lamby** [comment]
Great stuff!

**Pull request merged**